src/lib/selector/fast-auth-wallet.ts

```diff
@@ -0,0 +1,310 @@
+import { encodeSignedDelegate, SCHEMA, SignedDelegate } from '@near-js/transactions';
+import type {
+  Account,
+  BrowserWallet,
+  Network,
+  Optional,
+  Transaction,
+  WalletBehaviourFactory,
+  WalletModuleFactory,
+} from '@near-wallet-selector/core';
+import { createAction } from '@near-wallet-selector/wallet-utils';
+import * as nearAPI from 'near-api-js';
+import { deserialize } from 'near-api-js/lib/utils/serialize';
+
+import icon from './fast-auth-icon';
+import { FastAuthWalletConnection } from './fastAuthWalletConnection';
+export interface FastAuthWalletParams {
+  walletUrl?: string;
+  iconUrl?: string;
+  deprecated?: boolean;
+  successUrl?: string;
+  failureUrl?: string;
+  relayerUrl?: string;
+}
+
+interface FastAuthWalletState {
+  wallet: FastAuthWalletConnection;
+  keyStore: nearAPI.keyStores.BrowserLocalStorageKeyStore;
+  near: any;
+}
+
+interface FastAuthWalletExtraOptions {
+  walletUrl: string;
+  relayerUrl: string;
+}
+
+const processSignedDelegates = (relayerUrl: string, closeDialog: () => void, event: MessageEvent) => {
+  if (
+    event.data.signedDelegates &&
+    event.data.signedDelegates
+      .split(',')
+      .some((s: string) => deserialize(SCHEMA, SignedDelegate, Buffer.from(s, 'base64')))
+  ) {
+    closeDialog && closeDialog();
+    Promise.all(
+      event.data.signedDelegates.split(',').map((s: string) => {
+        const signedDelegate = deserialize(SCHEMA, SignedDelegate, Buffer.from(s, 'base64'));
+        return fetch(relayerUrl, {
+          method: 'POST',
+          mode: 'cors',
+          body: JSON.stringify(Array.from(encodeSignedDelegate(signedDelegate))),
+          headers: new Headers({ 'Content-Type': 'application/json' }),
+        });
+      }),
+    );
+    return true;
+  }
+  return false;
```

Comment:
this function includes events for posting request to relayer which is async, but it doesn't return the promise and instead it simply returns boolean without tracking the promise. Would it be nicer to return the promise?

Also should we be checking error handling when signed delegate failed on relayer? (which has happened multiple times already)

Reply:
Good catch, removed this function and handled the listener in `FastAuthWalletConnection`. Re error handling, I'm not sure how the VM handles those but I will address it in a separate PR as they are not currently handled for existing wallets